main.py

```diff
@@ -48,177 +62,352 @@ def validate_yaml_file(yaml_path: Path) -> dict:
 
         for key, expected_type in required_keys.items():
             if key not in parameters:
-                if key in ['company_blacklist', 'title_blacklist', 'location_blacklist']:
+                if key in [
+                    "company_blacklist",
+                    "title_blacklist",
+                    "location_blacklist",
+                ]:
                     parameters[key] = []
+                    logger.debug(f"Optional key '{key}' missing, set to empty list.")
                 else:
-                    raise ConfigError(f"Missing or invalid key '{key}' in config file {config_yaml_path}")
+                    logger.error(f"Missing key '{key}' in config file.")
+                    raise ConfigError(
+                        f"Missing or invalid key '{key}' in config file {config_yaml_path}"
+                    )
             elif not isinstance(parameters[key], expected_type):
-                if key in ['company_blacklist', 'title_blacklist', 'location_blacklist'] and parameters[key] is None:
+                if (
+                    key in ["company_blacklist", "title_blacklist", "location_blacklist"]
+                    and parameters[key] is None
+                ):
                     parameters[key] = []
+                    logger.debug(f"Key '{key}' was None, set to empty list.")
                 else:
-                    raise ConfigError(f"Invalid type for key '{key}' in config file {config_yaml_path}. Expected {expected_type}.")
-
-        # Validate experience levels, ensure they are boolean
-        experience_levels = ['internship', 'entry', 'associate', 'mid_senior_level', 'director', 'executive']
+                    logger.error(
+                        f"Invalid type for key '{key}' in config file. Expected {expected_type}, but got {type(parameters[key])}."
+                    )
+                    raise ConfigError(
+                        f"Invalid type for key '{key}' in config file {config_yaml_path}. Expected {expected_type}, but got {type(parameters[key])}."
+                    )
+
+        # Validate experience levels
+        experience_levels = [
+            "internship",
+            "entry",
+            "associate",
+            "mid_senior_level",
+            "director",
+            "executive",
+        ]
         for level in experience_levels:
-            if not isinstance(parameters['experience_level'].get(level), bool):
-                raise ConfigError(f"Experience level '{level}' must be a boolean in config file {config_yaml_path}")
-
-        # Validate job types, ensure they are boolean
-        job_types = ['full_time', 'contract', 'part_time', 'temporary', 'internship', 'other', 'volunteer']
+            if not isinstance(parameters["experience_level"].get(level), bool):
+                logger.error(
+                    f"Invalid value for experience level '{level}'. Expected a boolean (True/False)."
+                )
+                raise ConfigError(
```

Comment:
problem is we don't dump all errors at once. They fix error run it again to next validation errors.

raising expception will break for loop

Reply:
done